File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let [leftPtr, rightPtr] = [0, heights.length - 1];

        let maxArea = 0;
        while (leftPtr < rightPtr) {
            maxArea = Math.max(
                (rightPtr - leftPtr) * Math.min(heights[leftPtr], heights[rightPtr]),
                maxArea,
            );
            if (heights[leftPtr] < heights[rightPtr]) {
                leftPtr++
            } else if (heights[rightPtr] < heights[leftPtr]) {
                rightPtr--
            } else {
                leftPtr++
                rightPtr--
            }
        }

        return maxArea;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buyingPrice = prices[0]
        maxProfit = 0

        for i in range(1, len(prices)):
            maxProfit = max(prices[i] - buyingPrice, maxProfit)
            buyingPrice = min(buyingPrice, prices[i])

        return maxProfit
